nanoFramework.Tools.DebugLibrary.Shared/PortSerial/PortSerial.cs

```diff
@@ -282,13 +333,22 @@ public byte[] ReadBuffer(int bytesToRead)
 
                 try
                 {
-                    int readBytes = Device.Read(buffer, 0, bytesToRead);
+                    // this loop shall fix issues with M1 and some drivers showing up bytes to fast
+                    List<byte> received = new List<byte>();
+                    int readBytes = 0;
+                    while (Device.BytesToRead > 0 && readBytes < bytesToRead)

```

Comment:
Still missing a small change to make sure we're not reading more than what was requested.